client/my-sites/themes/themes-banner/index.jsx

```diff
@@ -3,34 +3,105 @@
 /**
  * External dependencies
  */
+
 import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+
+import { localize } from 'i18n-calypso';
 import PropTypes from 'prop-types';
-import cssSafeUrl from 'lib/css-safe-url';
 import Button from 'components/button';
+import safeImageUrl from 'lib/safe-image-url';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { getActiveTheme, getThemeDetailsUrl } from 'state/themes/selectors';
 
-export class ThemesBanner extends PureComponent {
+class ThemesBanner extends PureComponent {
 	static propTypes = {
-		title: PropTypes.string.isRequired,
-		description: PropTypes.string.isRequired,
-		action: PropTypes.func,
-		actionLabel: PropTypes.string.isRequired,
-		backgroundImage: PropTypes.string,
-		href: PropTypes.string,
+		themeId: PropTypes.string,
+		themeName: PropTypes.string,
+		title: PropTypes.node.isRequired,
+		description: PropTypes.node.isRequired,
+		buttonLabel: PropTypes.string.isRequired,
+		backgroundColor: PropTypes.string,
+		image: PropTypes.string,
+		imageAttrs: PropTypes.shape( {
+			width: PropTypes.number,
+		} ),
+		imageTransform: PropTypes.string,
+
+		// Connected props
+		siteId: PropTypes.number,
+		themeUrl: PropTypes.string,
+		activeThemeId: PropTypes.string,
+	};
+
+	onClick = () => {
```

Comment:
High-level nit: I've been calling my handlers like that 😄  until someone suggested I use a more descriptive name (to illustrate what the method actually does, not what it's used for). So maybe rename it to something like `recordEvent`?

Reply:
Interesting point, will update accordingly. Makes absolute sense!